src/components/accountForm/SignUp/SignUp.jsx

```diff
@@ -38,7 +38,7 @@ const SignUp = ({ header, onSubmit, newUserHandler, name = '', email = '', passw
         } else if (name.length > 40) {
             newErrors.name = 'Name is too long!';
         }
-        if (!email || email === '') {
+        if (!email || !/\S+@\S+\.\S+/.test(form.email)) {
             newErrors.email = 'E-mail is required!';
```

Comment:
this is wrong... you check if email exists (this is right)
but then.. at the same time you check if email format is valid and still displays that email is required. This message is nonsense